docs/malformed_files.md

```diff
@@ -0,0 +1,72 @@
+# Dealing with malformed files
+
+## Field compatibility
+
+When loading multiple files, the `INFO` and `FORMAT` fields
+from all VCF files are merged together to generate a "representative header",
+which is then used to generate a single BigQuery schema. If the same key is
+defined in multiple files, then its definition must be compatible across files.
+The compatibility rules are as follows:
+
+* Fields are compatible if they have the same `Number` and `Type` fields.
+  Annotation fields (i.e. those specified by `--annotation_fields`) must also
+  have the same `Description`.
+
+* Fields with different `Type` values are compatible in the following cases:
+
+  * `Integer` and `Float` fields are compatible and are converted to `Float`.
+  * We are adding more compatibility options (e.g. resolving `String` and
+  `Integer` to `String`). See
+  [below](#specifying---allow-incompatible-records) for more details.
+
+* Fields with different `Number` values are compatible in the following cases:
+
+  * "Repeated" numbers are compatible with each other. They include:
+    * `Number=.` (unknown number)
+    * Any `Number` greater than 1.
+    * `Number=G` (one value per genotype) and `Number=R` (one value for each
+      alternate and reference).
+    * `Number=A` (one value for each alternate) only if running with
+      `--split_alternate_allele_info_fields False`.
+
+  * We are adding more compatibility options (e.g. resolving `Number=1` and
+    `Number=.` with `Number=.`). See
+    [below](#specifying---allow-incompatible-records) for more details.
+
+
+## Specifying `--representative_header_file`
+
+The headers in the `--representative_header_file <path_to_file>` essentially
+specify the merged headers from all files being loaded to BigQuery. This file is
+used to directly generate the BigQuery schema. Note that we only read the
+header info from the file and ignore VCF records, so the
+`representative_header_file` can either be a file containing *just* the header
+fields or can point to an actual VCF file. Providing this file can be useful
+for:
+
+* Speeding up the pipeline especially if a large number of files are provided.
+  The pipeline will use the provided file to generate the BigQuery schema and
+  will skip merging headers across files. This is particularly useful if all
+  files have identical VCF headers.
+* Providing definitions for missing header fields. See the
+  [troubleshooting page](./troubleshooting.md) for more details.
+* Coming soon: resolving incompatible fields across files.
```

Comment:
Please update here too. We already do this and treat the number/type from the header fields regardless of what is provided in the file.

Reply:
Done